src/BuiltInTools/HotReloadAgent/MetadataUpdateHandlerInvoker.cs

```diff
@@ -327,15 +327,46 @@ internal static List<Assembly> TopologicalSort(Assembly[] assemblies)
 
         static void Visit(Assembly[] assemblies, Assembly assembly, List<Assembly> sortedAssemblies, HashSet<string> visited)
         {
-            var assemblyIdentifier = assembly.GetName().Name!;
+            string assemblyIdentifier;
+
+            try
+            {
+                assemblyIdentifier = assembly.GetName().Name!;
+            }
+            catch
+            {
+                return;
+            }
+
             if (!visited.Add(assemblyIdentifier))
             {
                 return;
             }
 
-            foreach (var dependencyName in assembly.GetReferencedAssemblies())
+            AssemblyName[] referencedAssemblies;
+            try
             {
-                var dependency = Array.Find(assemblies, a => a.GetName().Name == dependencyName.Name);
+                referencedAssemblies = assembly.GetReferencedAssemblies();
+            }
+            catch
+            {
+                referencedAssemblies = [];
+            }
+
+            foreach (var dependencyName in referencedAssemblies)
+            {
+                var dependency = Array.Find(assemblies, a =>
+                {
+                    try
+                    {
+                        return a.GetName().Name.Equals(dependencyName.Name, StringComparison.OrdinalIgnoreCase);
```

Comment:
The Name property on an AssemblyName is string?; would it be better to use string.Equals(a.GetName().Name, dependencyName.Name, StringComparision.ORdinalIgnoreCase))?

Reply:
Hmm...

![image](https://github.com/user-attachments/assets/dfc7e505-6d04-4047-a3ca-2267881587b3)